content/learn/book/ecs/entities-components/_index.md

```diff
@@ -0,0 +1,252 @@
++++
+title = "Entities have components"
+weight = 1
+template = "book-section.html"
+page_template = "book-section.html"
++++
+
+As we discussed in the introduction to this chapter, **entities** represent objects in your game world, whose data is stored in the form of components.
+
+The very first thing we're going to want to do is define the components that we'd like to use.
+To do so, we simply create Rust types (ensuring that they are `Send + Sync + 'static` so they can be sent across the threads safely) and pick a descriptive name for them.
+On this page, we're going to create a simple little combat system, so lets start by defining some basic components.
+
+```rust
+// These are dataless "unit structs", which hold no data of their own
+// In Bevy, these are useful for distinguishing similar entities or toggling behavior
+// and are called "marker components"
+struct Player;
+struct Enemy;
+
+// These simple components wrap an i8 in a tuple struct
+struct Life(u8);
+struct Attack(u8);
+struct Defense(u8);
+
+// We can store arbitrary data in our components, as long as it has a 'static lifetime
+struct Name(String);
+```
+
+`Life`, `Attack` and `Defense` will almost always be added to our entities at the same time, so let's create a **bundle** (collection of components) to make them easier to work with.
+
+```rust
+#[derive(Bundle)]
+struct CombatBundle {
+	// Each field of this bundle corresponds to a type of component 
+	// that will be inserted into the final entity
+	// The field names don't matter: only the types are retained in our ECS storage
+    life: Life,
+    attack: Attack,
+    defense: Defense,
+}
+```
+
+Now, let's get to work by spawning a player entity and an enemy entity for them to fight!
```

Comment:
I think that this section could end here. After this point there is a lot of talk involving systems, that should not belong in the "entities have components" section. I think we shouldn't create the whole game in one single section. We can instead introduce each concept as soon as it's needed during the development.

First, I would remove the player input from the example (basically it becomes a zero player game), and then have the next sections as the following:
1. _Commands._ Introduce commands, like in the already existing section, then use the acquired knowledge to spawn the player and the enemy in a startup system.
2. _Queries._ Introduce queries. Show an example of how a query looks in the context of a system parameter, then link to the API docs for the full usage. Then add the update systems. At this point, the `NextAttack` resource doesn't exist yet, so make the player relentlessly attack the enemy until their hp goes to 0. 
3. _Resources._ Introduce resources. Add the `NextAttack` resource, and make the poor enemy defend themselves.

For each section, after enriching the example, I would continue, when appropriate, by talking about less common use cases of the concept, connecting with previous topics and so on.

I left out the dedicated section for systems. I don't know what we should do with that: systems pervade everything, and making an example starting with systems will inevitably bleed into other areas, like commands, queries and resources. On the other hand, a prior discussion of systems is needed to explain their role in relation to the `World` (the fact that they can read and mutate data) and to the `Schedule` (they can be set to run once or for each iteration of the game loop). Therefore, I think that introducing systems before the example could be the way to go, without obviously mentioning queries, commands or resources.